From: Takashi Iwai <tiwai@suse.de>
Subject: [ALSA] hdsp: always initialize card name
Patch-mainline: 
References: 164962

When using the kernel firmware loader, initialize the card shortname
before registering the card with ALSA to prevent it from using a
default card ID.

Signed-off-by: Clemens Ladisch <clemens@ladisch.de>
Acked-by: Takashi Iwai <tiwai@suse.de>

---


diff -u linux/sound/pci/rme9652/hdsp.c:1.92 linux/sound/pci/rme9652/hdsp.c:1.93
--- linux/sound/pci/rme9652/hdsp.c:1.92	Wed Aug  3 05:50:31 2005
+++ linux/sound/pci/rme9652/hdsp.c	Fri Sep  2 00:49:21 2005
@@ -4899,6 +4899,7 @@
 	}
 	
 	if (!(hdsp->state & HDSP_InitializationComplete)) {
+		strcpy(card->shortname, "Hammerfall DSP");
 		sprintf(card->longname, "%s at 0x%lx, irq %d", hdsp->card_name, 
 			hdsp->port, hdsp->irq);
 	    
